microsoft/testsuites/rust_vmm_mshv/rust_vmm_mshv_test.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import re
+from pathlib import Path
+
+from assertpy.assertpy import assert_that
+
+from lisa import (
+    Environment,
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    notifier,
+)
+from lisa.messages import SubTestMessage, TestStatus, create_test_result_message
+from lisa.testsuite import TestResult
+from lisa.tools import Cargo, Git
+from lisa.util.process import ExecutableResult
+
+
+@TestSuiteMetadata(
+    area="rust-vmm",
+    category="community",
+    description="""
+    This test suite is for executing the rust-vmm/mshv tests
+    """,
+)
+class RustVmmTestSuite(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            Runs rust-vmm/mshv tests
+        """,
+        priority=3,
+        timeout=1800,
+    )
+    def verify_rust_vmm_mshv_tests(
+        self,
+        log: Logger,
+        node: Node,
+        environment: Environment,
+        log_path: Path,
+        result: TestResult,
+    ) -> None:
+        repo = "https://github.com/rust-vmm/mshv.git"
+        git = node.tools[Git]
+        repo_root = git.clone(repo, node.get_working_path())
+        testcase_log = log_path.joinpath("rust_vmm_mshv.log")
+        cargo = node.tools[Cargo]
+        test_result: ExecutableResult = cargo.test(cwd=repo_root, sudo=True)
+        with open(testcase_log, "w") as f:
+            f.write(f"{test_result.stdout} {test_result.stderr}")
+        self.__process_result(
+            test_result.stdout,
+            result,
+            environment,
+        )
+
+    def __process_result(
+        self,
+        data: str,
+        result: TestResult,
+        environment: Environment,
+    ) -> None:
+        pattern = r"\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])"
+        ansi_escape = re.compile(pattern)
+        data = ansi_escape.sub("", data)
+        match = re.findall(
+            r"test (.*?) ... (ok|ignored|failed)(.*?)\n",
+            data,
+            re.IGNORECASE,
+        )
+        failed_testcases = []
+        for testcase in match:
+            status = TestStatus.QUEUED
+            testcase_name = testcase[0]
+            log_status = testcase[-2]
+            if log_status and log_status.strip().lower() == "ok":
+                status = TestStatus.PASSED
+            elif log_status and log_status.strip().lower() == "failed":
+                failed_testcases.append(testcase_name)
+                status = TestStatus.FAILED
+            elif log_status and log_status.strip().lower() == "ignored":
+                status = TestStatus.SKIPPED
+            self.__send_subtest_msg(
+                result,
+                environment,
+                testcase_name,
+                status,
+            )
+        assert_that(
+            failed_testcases, f"Failed Testcases: {failed_testcases}"
```

Comment:
Not sure the frequency of failures and the logs generated. But is it possible to have separate log files for individual subtest failures? So that its easier for debugging why a subtest failed.

Reply:
We cannot run subtest individually. It is testcase framework which run it with single command. I am already collecting log file for entire testcase run. You can refer https://github.com/rust-vmm/mshv